Sources/Shared/Extensions/Dictionary+Tailor.swift

```diff
@@ -0,0 +1,41 @@
+import Sugar
+
+public extension Dictionary {
+
+  func propertyOrThrow<T>(name: String) throws -> T {
+    guard let key = name as? Key,
+      value = self[key] as? T
+      else { throw MappableError.TypeError(message: "Tried to get value for \(name) as \(T.self)") }
+
+    return value
+  }
+
+  func property<T>(name: String) -> T? {
+    guard let key = name as? Key,
+      value = self[key]
+      else { return nil }
+
+    return value as? T
+  }
+
+  func transform<T, U>(name: String, transformer: ((value: U?) -> T?)) -> T? {
+    guard let value = self[name as! Key] else { return nil }
+    return transformer(value: value as? U)
+  }
+
+  func relation<T : Mappable>(name: String) -> T? {
```

Comment:
@zenangst What about 
`func property<T: Mappable>(name: String) -> T?` 
`func property<T : Mappable>(name: String) -> [T]?`

Like we discussed today, the same name, but for multiple possible types.